elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchHSQueryImpl.java

```diff
@@ -694,6 +696,16 @@ private SearchResult search(boolean enableScrolling) {
 				}
 			}
 			Search search = builder.build();
+
+			if ( QUERY_LOG.isDebugEnabled() ) {
+				try ( ServiceReference<GsonService> gsonService =
+						extendedIntegrator.getServiceManager().requestReference( GsonService.class ) ) {
```

Comment:
I don't think I can:

 * those queries are created in `org.hibernate.search.elasticsearch.impl.ElasticsearchJsonQueryDescriptor.createHSQuery(SearchIntegrator)`, and that method is generic SPI. We could change it, but we certainly couldn't (shouldn't) add Elasticsearch-specific parameters.
 * the query descriptors themselves may be created from public static methods (such as `org.hibernate.search.elasticsearch.ElasticsearchQueries.fromJson(String)`), making it impossible to pass a reference to the service when creating the query descriptor.

What I can do is:

 * request the reference only once in the `ElasticsearchHSQuery` constructor, which is not much better since queries are generally only executed once or twice. 
 * replace the GsonService by static methods, but I did exactly the opposite a few months ago, so that would be a shame.
 * maybe see why requesting a reference to a service is such a performance issue? Especially given we have a cache...

Also, I'd like to mention two facts:

 1. We already request a reference to a service each time we create a query through `org.hibernate.search.engine.impl.ImmutableSearchFactory.createHSQuery(Query, Class<?>...)` (provided Elasticsearch is being used)
 2. This code is only executed in debug mode, which I'd expect to be less efficient than normal mode.

What do you think?